relayer/strategies.go

```diff
@@ -1,58 +1,241 @@
 package relayer
 
 import (
+	"context"
 	"fmt"
+	"os"
+	"strings"
 	"time"
+
+	"github.com/avast/retry-go"
+	"github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
 )
 
-// StartRelayer starts the main relaying loop
+// ActiveChannel represents an IBC channel and whether there is an active goroutine relaying packets against it.
+type ActiveChannel struct {
+	channel *types.IdentifiedChannel
+	active  bool
+}
+
+// StartRelayer starts the main relaying loop.
 func StartRelayer(src, dst *Chain, maxTxSize, maxMsgLength uint64) (func(), error) {
 	doneChan := make(chan struct{})
+	channels := make(chan *ActiveChannel, 10)
+	var srcOpenChannels []*ActiveChannel
+
 	go func() {
 		for {
 			select {
 			case <-doneChan:
 				return
 			default:
-				// Fetch any unrelayed sequences depending on the channel order
-				sp, err := UnrelayedSequences(src, dst)
+				// Query the list of channels on the src connection
+				srcChannels, err := QueryChannelsOnConnection(src)
 				if err != nil {
-					src.Log(fmt.Sprintf("unrelayed sequences error: %s", err))
-				} else {
-					if len(sp.Src) > 0 && src.debug {
-						src.Log(fmt.Sprintf("[%s] unrelayed-packets-> %v", src.ChainID(), sp.Src))
-					}
-					if len(sp.Dst) > 0 && dst.debug {
-						dst.Log(fmt.Sprintf("[%s] unrelayed-packets-> %v", dst.ChainID(), sp.Dst))
-					}
-					if !sp.Empty() {
-						if err = RelayPackets(src, dst, sp, maxTxSize, maxMsgLength); err != nil {
-							src.Log(fmt.Sprintf("relay packets error: %s", err))
-						}
-					}
+					fmt.Printf("error querying all channels on chain{%s}@connection{%s}: %v \n",
```

Comment:
I would like to see this remain a logger call, or change the method to accept a `stderr io.Writer` argument to use with `fmt.Fprintf` here, but that can change later.